wsd/ProxyRequestHandler.cpp

```diff
@@ -0,0 +1,99 @@
+/* -*- Mode: C++; tab-width: 4; indent-tabs-mode: nil; c-basic-offset: 4; fill-column: 100 -*- */
+/*
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/.
+ */
+
+#include <config.h>
+
+#include <Poco/URI.h>
+
+#include <COOLWSD.hpp>
+#include "ProxyRequestHandler.hpp"
+#include <net/HttpRequest.hpp>
+#include <net/HttpHelper.hpp>
+
+std::map<unsigned long, std::shared_ptr<StreamSocket>> ProxyRequestHandler::LinkSession;
+std::map<std::string, std::shared_ptr<http::Response>> ProxyRequestHandler::CacheFileHash;
+std::chrono::system_clock::time_point ProxyRequestHandler::MaxAge;
+
+void ProxyRequestHandler::handleRequest(const std::string& relPath,
+                                        const std::shared_ptr<StreamSocket>& socket)
+{
+    Poco::URI uriProxy(ProxyServer);
+    const auto timeNow = std::chrono::system_clock::now();
+
+    if (MaxAge > std::chrono::system_clock::time_point() && timeNow > MaxAge)
+    {
+        CacheFileHash.clear();
+        MaxAge = std::chrono::system_clock::time_point();
+    }
+
+    const auto cacheEntry = CacheFileHash.find(relPath);
+    if (cacheEntry != CacheFileHash.end())
+    {
+        socket->sendAndShutdown(*cacheEntry->second.get());
```

Comment:
`shared_ptr` has a very friendly `operator*` exactly for this case. Use `*cacheEntry->second` (no need to call `.get()`). Same pattern applies elsewhere in this PR.